sycl/test-e2e/KernelCompiler/kernel_compiler_spirv.cpp

```diff
@@ -0,0 +1,158 @@
+//==- kernel_compiler_spirv.cpp --------------------------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// REQUIRES: ocloc
+
+// RUN: %clang -c -target spir64 -O0 -emit-llvm %S/Kernels/spirv_tests.cl -o %t.bc && llvm-spirv %t.bc -o %t.spv
+// RUN: %clang -DENABLE_FP16 -c -target spir64 -O0 -emit-llvm %S/Kernels/spirv_tests.cl -o %t_fp16.bc && llvm-spirv %t_fp16.bc -o %t_fp16.spv
+// RUN: %clang -DENABLE_FP64 -c -target spir64 -O0 -emit-llvm %S/Kernels/spirv_tests.cl -o %t_fp64.bc && llvm-spirv %t_fp64.bc -o %t_fp64.spv
+// RUN: %{build} -o %t.out
+// RUN: %{run} %t.out %t.spv %t_fp16.spv %t_fp64.spv
+
+// Test case for the sycl_ext_oneapi_kernel_compiler_spirv extension. This test
+// loads pre-compiled kernels from a SPIR-V file and runs them.
+
+#include <array>
+#include <cassert>
+#include <fstream>
+#include <string>
+#include <sycl/sycl.hpp>
+
+sycl::kernel_bundle<sycl::bundle_state::executable>
+loadKernelsFromFile(sycl::queue &q, std::string file_name) {
+  namespace syclex = sycl::ext::oneapi::experimental;
+
+  // Read the SPIR-V module from disk.
+  std::ifstream spv_stream(file_name, std::ios::binary);
+  spv_stream.seekg(0, std::ios::end);
+  size_t sz = spv_stream.tellg();
+  spv_stream.seekg(0);
+  std::vector<std::byte> spv(sz);
+  spv_stream.read(reinterpret_cast<char *>(spv.data()), sz);
+
+  // Create a kernel bundle from the binary SPIR-V.
+  sycl::kernel_bundle<sycl::bundle_state::ext_oneapi_source> kb_src =
+      syclex::create_kernel_bundle_from_source(
+          q.get_context(), syclex::source_language::spirv, spv);
+
+  // Build the SPIR-V module for our device.
+  sycl::kernel_bundle<sycl::bundle_state::executable> kb_exe =
+      syclex::build(kb_src);
```

Comment:
`sycl::build` also takes a property list for logging and build options. Are those supported by SPIR-V as well? If so we should test them.